draft-ietf-httpbis-messaging-latest.xml

```diff
@@ -685,50 +685,45 @@ Host: www.example.org:8001
 
 <section title="Reconstructing the Target URI" anchor="reconstructing.target.uri">
   <x:anchor-alias value="h1.effective.request.uri"/>
-<t>
-   Since the request-target often contains only part of the user agent's
-   target URI, a server constructs its value to properly service the request
-   (<xref target="target.resource"/>).
-</t>
 <t>
    If the <x:ref>request-target</x:ref> is in <x:ref>absolute-form</x:ref>,
    the target URI is the same as the request-target. Otherwise, the
-   target URI is constructed as follows:
+   server reconstructs the target URI from the connection context and various
+   parts of the request message in order to identify the target resource
+   (<xref target="target.resource"/>):
 </t>
 <ul>
-<li>
+<li><t>
    If the server's configuration (or outbound gateway) provides a fixed URI
    scheme, that scheme is used for the target URI.
    Otherwise, if the request is received over a secured connection,
    the target URI's scheme is "https"; if not, the scheme is "http".
-</li>
-<li>
-   If the server's configuration (or outbound gateway) provides a fixed URI
-   <x:ref>authority</x:ref> component, that authority is used for the
-   target URI. If not, then if the request-target is in
-   <x:ref>authority-form</x:ref>, the target URI's authority
-   component is the same as the request-target.
-   If not, then if a <x:ref>Host</x:ref> header field is supplied with a
-   non-empty field value, the authority component is the same as the
-   Host field value. Otherwise, the authority component is assigned
-   the default name configured for the server and, if the connection's
-   incoming TCP port number differs from the default port for the target
-   URI's scheme, then a colon (":") and the incoming port number (in
-   decimal form) are appended to the authority component.
+</t></li>
+<li><t>
+   If the request-target is in <x:ref>authority-form</x:ref>,
+   the target URI's authority component is the same as the request-target.
+   Otherwise, the request-target is in <x:ref>origin-form</x:ref> and:
```

Comment:
Where are the asterisk-form clauses for determining authority?

I think that you have a three-fold conditional here, which this presentation doesn't seem to work quite right for:

1. If authority-form, authority is right there.
2. If origin-form AND Host is valid, authority is the value of Host.
3. Otherwise, the authority component is empty.

It looks like you could write that as three sentences without the extra nested list.

Though do you need the "if origin-form" condition on the second item?  Can the same apply to asterisk-form?  That is, condition 2 is "If Host is valid, authority is the value of Host."  The `OPTIONS *` example below suggests that it isn't necessary.

Incidentally, I initially read "Otherwise, the request is in origin-form" as a statement of fact rather than a condition, which was very confusing at first.

Reply:
You are probably right, but I need sleep in order to think it through again. Will try again tomorrow.

Reply:
Right, done in https://github.com/httpwg/http-core/commit/0cf513d7c8119c3cac1db863d00d24684d37c4ff